src/Board.java

```diff
@@ -150,19 +150,21 @@ private boolean kingFound() {
         return false;
     }
 
-    public int evaluate(Pion pion) {
+    public int evaluate(boolean isCurrentPlayer, Pion currentPlayer) {
+        Pion pion = isCurrentPlayer ? currentPlayer : Pion.getOppositePion(currentPlayer);
+
```

Comment:
why make this more complicated than it was?

Reply:
Because it would always go to the evaluate of the cpu instead of evaluating for red, then black, ...

Reply:
Ah ok I see

Reply:
> Because it would always go to the evaluate of the cpu instead of evaluating for red, then black, ...

Wait actually no, we actually want it to evaluate always from the cpu's perpective.

Reply:
sure
opir on le modifiera plus tard

Reply:
Well no we will not modify it